cmd/metricsbridge/metricsbridge.go

```diff
@@ -235,6 +243,20 @@ func (c *config) runOnce(ctx context.Context) error {
 					f.Dims[string(k)] = string(v)
 				}
 			}
+			//if there's an instance dimension try to lookup the hostname
+			if instance, found := sample.Metric["instance"]; found {
+				hostname, present := hostnameMap[strings.Split(string(instance), ":")[0]]
+				if present {
+					//only add the hostname dimension if the lookup succeeds
+					f.Dims["hostname"] = hostname
+					hostnameprefix := strings.Split(hostname, "-")[0]
```

Comment:
belated request: please could you put the function that calculates the agent pool profile role from the hostname into pkg/config/names.go with a unit test? that way we keep all our expectations about how these things map together into one place?

Reply:
Moved function to GetAgentRole in pkg/config/names.go, added unit tests.